<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <xacro:macro name="diffbot_system" params="name">
    <ros2_control name="${name}" type="system">
      <hardware>
        <plugin>ros_phoenix/PhoenixSystem</plugin>

        <param name="interface">can0</param>
      </hardware>

      <joint name="left_wheel_joint">
        <param name="type">ros_phoenix::TalonFX</param>
        <param name="id">20</param>
        <param name="edges_per_rot">2048</param>
        <param name="sensor_multiplier">0.00048828125</param>
        <param name="P">0.08</param>
        <param name="I">0.0</param>
        <param name="D">0.0</param>
        <param name="F">0.04702194357</param>

        <command_interface name="velocity"/>

        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <joint name="right_wheel_joint">
        <param name="type">ros_phoenix::TalonFX</param>
        <param name="id">21</param>
        <param name="edges_per_rot">2048</param>
        <param name="sensor_multiplier">0.00048828125</param>
        <param name="P">0.08</param>
        <param name="I">0.0</param>
        <param name="D">0.0</param>
        <param name="F">0.04702194357</param>

        <command_interface name="velocity"/>

        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
    </ros2_control>
  </xacro:macro>
</robot>